constitution/actions.js

```diff
@@ -1204,4 +1245,40 @@ const actions = new Map([
       function (args) {}
     ),
   ],
+  [
+    "set_public_prefix",
+    new Action(
+      // validate
+      function (args) {
+        // TODO: this should probably be a base64 string so it can represent full byte vectors
+        checkType(args.public_prefix, "string", "public_prefix");
+        const public_prefix_map = ccf.kv[publicPrefixMapName];
+        if (public_prefix_map.has(ccf.strToBuf(args.public_prefix))) {
+          throw new Error("Public prefix already set");
+        }
```

Comment:
We usually try to make our proposal idempotent, and I don't think there's a strong reason why we should throw here?

Reply:
Yeah, wasn't too sure about this and idempotency does seem like a better way to go.